compiler/rustc_mir/src/borrow_check/type_check/mod.rs

```diff
@@ -1520,6 +1520,7 @@ impl<'a, 'tcx> TypeChecker<'a, 'tcx> {
                     );
                 }
             }
+            StatementKind::CopyNonOverlapping(..) => todo!(),
```

Comment:
The `check_call_inputs` should be a good template to utilize here.

Reply:
I've intentionally not resolved this one, as I'm not sure if my implementation is correct. Specifically, I was extra lazy and instead of passing an additional type parameter, I just compare the two arguments that they are of equal types, which I think is sufficient.